src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.Netstandard2_0.cs

```diff
@@ -0,0 +1,628 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.DependencyModel
+{
+    public partial class DependencyContextJsonReader : IDependencyContextReader
+    {
+        public DependencyContext Read(Stream stream)
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+
+            ArraySegment<byte> drained = ReadToEnd(stream);
+
+            DependencyContext dependencyContext;
+            try
+            {
+                dependencyContext = ReadCore(drained.AsSpan(0, drained.Count));
+            }
+            catch
+            {
+                // Holds document content, clear it before returning it.
+                drained.AsSpan(0, drained.Count).Clear();
+                ArrayPool<byte>.Shared.Return(drained.Array);
+                throw;
+            }
+            finally
+            {
+                // Holds document content, clear it before returning it.
+                drained.AsSpan(0, drained.Count).Clear();
+                ArrayPool<byte>.Shared.Return(drained.Array);
+            }
+            return dependencyContext;
```

Comment:
Why the complication of defining the dependencyContext outside of the try/finally?  Why not just:
``C#
ArraySegment<byte> drained = ReadToEnd(stream);
try
{
    return ReadCore(drained.AsSpan());
}
finally
{
    drained.AsSpan().Clear();
    ArrayPool<byte>.Shared.Return(drained.Array);
}
```
?